src/common/ScreenExporter.hpp

```diff
@@ -0,0 +1,58 @@
+/* *****************************************************************************
+ * A.L.E (Arcade Learning Environment)
+ * Copyright (c) 2009-2013 by Yavar Naddaf, Joel Veness, Marc G. Bellemare and 
+ *   the Reinforcement Learning and Artificial Intelligence Laboratory
+ * Released under the GNU General Public License; see License.txt for details. 
+ *
+ * Based on: Stella  --  "An Atari 2600 VCS Emulator"
+ * Copyright (c) 1995-2007 by Bradford W. Mott and the Stella team
+ *
+ * *****************************************************************************
+ *  ScreenExporter.hpp 
+ *
+ *  A class for exporting Atari 2600 frames as PNGs.
+ *
+ **************************************************************************** */
+
+#ifndef __SCREEN_EXPORTER_HPP__
+#define __SCREEN_EXPORTER_HPP__ 
+
+#include <string>
+#include "display_screen.h"
+#include "ale_screen.hpp"
+
+class ScreenExporter {
+
+    public:
+
+        /** Creates a new ScreenExporter which can be used to save screens using save(filename). */ 
+        ScreenExporter(ColourPalette &palette);
```

Comment:
Ahh. Somehow I completely forgot you'd already implemented record_screen_dir, i.e. did all my listed tasks for me. Apologies for stomping on some of your work.

At first I was going to agree with you about letting StellaEnvironment do the saving. But now I'm realizing this doesn't play well with agents that save and load states.

I think we should provide the ScreenExporter object and drop saveScreenPNG from the interface. It might look a bit weird but it encapsulates the functionality better than if it were inside StellaEnvironment, which is about providing a proper RL environment. But I'm open to arguments.